### DISCUSSION TITLE
sending http request in js

### DISCUSSION THREAD

**Discussion by nikhil-varma-mudunuri** [pocketbase/pocketbase#3050] [Q&A] [answered]
how can i send post request to another API using JavaScript in pocketbase for every 2 mins

**Comment by ganigeorgiev** [comment, marked as answer]
You could check https://pocketbase.io/docs/js-sending-http-requests/.

If you want to execute something periodically, then you need to run it inside https://pocketbase.io/docs/js-jobs-scheduling.

For example:

```js
cronAdd("hello", "*/2 * * * *", () => {
    try {
        const res = $http.send({
            url:    "https://example.com",
            method: "POST",
            data:   {
                "field1": "a",
                "field2": 123,
                "field3": false,
            },
        })

        console.log(res.statusCode) // the response HTTP status code
        console.log(res.raw)        // the response body as plain text
        console.log(res.json)       // the response body as parsed json array or map
    } catch (err) {
        console.log("request failed", err);
    }
})
```

_Note that the current the JSVM `$http` client doesn't support `multipart/form-data` requests. Depending on the users demand it may be added in the future._

**Comment by nikhil-varma-mudunuri** [comment]
thanks for very fast responses.
